src/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs

```diff
@@ -32,19 +32,29 @@ public void TestEnvironmentProperty()
 
             IDictionary<string, string> environment = psi.Environment;
 
-            Assert.NotEqual(environment.Count, 0);
+            Assert.NotEqual(0, environment.Count);
 
-            int CountItems = environment.Count;
+            int countItems = environment.Count;
 
             environment.Add("NewKey", "NewValue");
             environment.Add("NewKey2", "NewValue2");
 
-            Assert.Equal(CountItems + 2, environment.Count);
+            Assert.Equal(countItems + 2, environment.Count);
             environment.Remove("NewKey");
-            Assert.Equal(CountItems + 1, environment.Count);
+            Assert.Equal(countItems + 1, environment.Count);
 
-            //Exception not thrown with invalid key
-            Assert.Throws<ArgumentException>(() => { environment.Add("NewKey2", "NewValue2"); });
+            // The full .NET framework represents Environment with StringDictionary. On .NET Core we use
+            // use Dictionary<string, string>, because we need to have case sensitive behaviour on Unix.
```

Comment:
We could definitely match Desktop behavior with StringDictionaryWrapper by making it implement IDictionary<TKey, TValue> and have its contents to be a Dictionary as it is, and when calling .Add instead of calling contents.Add() just do contents[key] so that we override the value instead of throwing. Being said that we would just need to create the StringDictionaryWrapper with a comparer to be passed to the Dictionary depending on the OS we are running on. 

Then for `Environment` property we could just return `EnvironmentVariables` as it would implement IDictionary.

We would need to use StringDictionary internals if we wanted to use/replace StringDictionary's contents which are a Hashtable and they are overriden in case it needs a comparer, but if we override IDictionary behavior by implementing it we could make it match and use our own contents to be a Dictionary. 

@Hughbe, can you do that?